### PR TITLE
Add support for reneg-sec option

### PR DESCRIPTION
**Pull request by oc243**

### CODE REVIEW

**Comment by oc243** [comment]
@luxflux any chance of a review of this when you get some time pls?

File: manifests/client.pp [hunk 1]
```diff
@@ -214,6 +218,7 @@
   $shared_ca            = undef,
   $custom_options       = {},
   $expire               = undef,
+  $reneg_sec            = '',
```
Comment (luxflux) [inline manifests/client.pp]: This does not match the documentation and IMHO a default value of `undef` is more consistent than an empty string. Could you change this?

File: manifests/server.pp [hunk 2]
```diff
@@ -491,6 +495,7 @@
   $tls_auth                  = false,
   $tls_server                = false,
   $tls_client                = false,
+  $reneg_sec                 = '',
```
Comment (luxflux) [inline manifests/server.pp]: Same here.

File: spec/defines/openvpn_client_spec.rb [hunk 3]
```diff
@@ -114,6 +115,7 @@
     it { should contain_file('/etc/openvpn/test_server/download-configs/test_client/test_client.conf').with_content(/^verb\s+1$/)}
     it { should contain_file('/etc/openvpn/test_server/download-configs/test_client/test_client.conf').with_content(/^mute\s+10$/)}
     it { should contain_file('/etc/openvpn/test_server/download-configs/test_client/test_client.conf').with_content(/^auth-retry\s+interact$/)}
+    it { should contain_file('/etc/openvpn/test_server/download-configs/test_client/test_client.conf').with_content(/^reneg-sec\s+3600$/)}
```
Comment (luxflux) [inline spec/defines/openvpn_client_spec.rb]: Please also add the reverse test (it should not be there by default).

File: spec/defines/openvpn_server_spec.rb [hunk 4]
```diff
@@ -211,6 +212,7 @@
 
     it { should contain_file('/etc/openvpn/test_server.conf').with_content(%r{^fragment 1412$}) }
     it { should contain_file('/etc/openvpn/test_server.conf').with_content(%r{^port-share 127.0.0.1 8443$}) }
+    it { should contain_file('/etc/openvpn/test_server.conf').with_content(/^reneg-sec\s+3600$/) }
```
Comment (luxflux) [inline spec/defines/openvpn_server_spec.rb]: Please also add the reverse test (it should not be there by default).

**Comment by oc243** [comment]
Hi @luxflux, thanks for the review. I think that the changes that I made to this branch on 13 Feb have addressed your points. Please may you take another look and see if they look good to you?

Thanks!

File: manifests/client.pp [hunk 5]
```diff
@@ -136,6 +136,10 @@
 #   days the certificate is valid for.
 #   Default: undef
 #
+# [*reneg_sec*]
+#   Integer. Value for how often to regenotiate keys.
```
Comment (actatux) [inline manifests/client.pp]: Typo: renegotiate

File: manifests/server.pp [hunk 6]
```diff
@@ -305,6 +305,10 @@
 #   Boolean. Allows you to set this server up as a tls-client connection.
 #   Default: false
 #
+# [*reneg_sec*]
+#   Integer. Value for how often to regenotiate keys.
```
Comment (actatux) [inline manifests/server.pp]: Typo: renegotiate

**Comment by bastelfreak** [comment]
Hi @oc243, can you please rebase this PR?

**Comment by bastelfreak** [comment]
Hi. I'm going to close this PR due to inactivity. Please reopen and rebase it if you're still interested in it.

**Pull request closed without merging**